src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json

```diff
@@ -53,41 +58,107 @@
       "replaces": "net8.0",
       "defaultValue": "net9.0"
     },
-    "AspireVersion": {
+    "AspireVersionCli": {
+      "type": "parameter",
+      "description": "The version of .NET Aspire to use.",
+      "displayName": ".NET Aspire version",
+      "datatype": "choice",
+      "isEnabled": "(hostIdentifier == \"dotnetcli\")",
+      "choices": [
+        {
+          "choice": "9.1",
+          "displayName": "9.1",
+          "description": "Chooses .NET Aspire 9.1"
+        },
+        {
+          "choice": "9.0",
+          "displayName": "9.0",
+          "description": "Chooses .NET Aspire 9.0"
+        },
+        {
+          "choice": "8.2",
+          "displayName": "8.2",
+          "description": "Chooses .NET Aspire 8.2 (only valid when '--framework net8.0' is specified)"
+        }
+      ],
+      "defaultValue": "9.1"
+    },
+    "AspireVersionNet8": {
       "type": "parameter",
       "description": "The version of .NET Aspire to use.",
       "displayName": ".NET Aspire version",
       "datatype": "choice",
       "isEnabled": "Framework == net8.0",
```

Comment:
I think you can remove the `AspireVersionCli` symbol entirely if you loosen this enablement condition to allow `Framework` to be either `net8.0` or `net9.0`. I think this is why `--framework net9.0 --aspire-version 9.0` doesn't actually impact the aspire version to be used.

Reply:
Check my latest commit for what I think *should* work for VS and CLI, but only works in VS. The commit before that is what I had to do to make it work in both.